.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,47 @@
+# Github actions that run tests on each commit, and publish to PyPi on tagged releases
+# build using https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+
+name: Test and Release to PyPi
+
+on:
+  push
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install and build
+      run: >-
+        python -m
+        pip install ".[dev]"
+        build
+        --user
+    - name: build binary and tarball
+      run: >-
+        python -m
+        build
+        --sdist
+        --wheel
+        --outdir dist/
+        .
+    - name: test
+      run: >-
+        python -m
+        pip install --upgrade pytest
+        pytest
```

Comment:
Could we also pass the relevant code coverage flags to pytest and run the `black` formatting checker like https://github.com/aws/aws-jupyter-proxy/blob/master/buildspec.yml#L11-L12 ?

Reply:
Yes, this is a good point, will add those back